core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -1135,6 +1132,10 @@ class KafkaServer(
    * @return The brokerId.
    */
   private def getOrGenerateBrokerId(metaPropsEnsemble: MetaPropertiesEnsemble): Int = {
+    if (config.migrationEnabled && config.brokerIdGenerationEnable) {
```

Comment:
The existing docs should be OK here, now that I have altered the PR to forbid broker id being set to -1, rather than forbidding `broker.id.generation.enable`.